py-pkg/pkgnet/abstract_graph_reporter.py

```diff
@@ -0,0 +1,92 @@
+from abc import abstractmethod
+
+from pkgnet.abstract_package_reporter import AbstractPackageReporter
+from pkgnet.graph_viz import VisJs
+from pkgnet.html_dependencies import HtmlDependencies
+
+
+class AbstractGraphReporter(AbstractPackageReporter):
+
+    _graph_class = None
+    _html_dependencies = HtmlDependencies(
+        scripts=["jquery-3.4.1.min.js", "datatables.min.js"], stylesheets=["datatables.min.css"]
+    )
+
+    def __init__(self, viz_class=VisJs):
+        super().__init__()
+        self._nodes = None
+        self._edges = None
+        self._pkg_graph = None
+        self._graph_viz = None
+        self._viz_class = viz_class
+
+    ### PUBLIC METHODS ###
+    def calculate_default_measures(self):
+        self.pkg_graph.node_measures(measures=self.pkg_graph.default_node_measures)
+        return self
+
+    def get_summary_view(self):
+        datatables_init_script = f"""
+        <script>
+            $(document).ready( function () {{
+                $('#{self.report_slug}-table').DataTable();
+            }} );
+        </script>
+        """
+        return (
+            self.nodes.to_html(classes=["display"], table_id=f"{self.report_slug}-table")
+            + datatables_init_script
+        )
+
+    ### PROPERTIES ###
+
+    @property
+    def nodes(self):
+        if self._nodes is None:
+            self._extract_nodes_and_edges()
+        return self._nodes
+
+    @property
+    def edges(self):
+        if self._edges is None:
+            self._extract_nodes_and_edges()
+        return self._edges
+
+    @property
+    def network_measures(self):
+        pass
```

Comment:
I think we should raise a `NotImplementtedError` instead of passing, to catch cases where a child class doesn't implement this

Reply:
Whoops, forgot an `@abstractmethod` here.